### PR TITLE
Useful to know more about high-resolution displays.

### LINKED ISSUE #18: Add getScaleFactor for widget and window
Links:
https://developer.gnome.org/gtk3/stable/GtkWidget.html#gtk-widget-get-scale-factor
https://developer.gnome.org/gdk3/stable/gdk3-Windows.html#gdk-window-get-scale-factor

### PR DESCRIPTION
**Pull request by tinchodias**
Fix #18.

I implemented them when investigated about Hiedra in a GTK window in a Apple Retina screen. 

Useful? else I just delete them.

Tell me if you want some test to be written,I can do it if not too difficult.

### CODE REVIEW

File: src/Mars-Gtk/GtkWidget.class.st [hunk 1]
```diff
@@ -462,6 +462,13 @@ GtkWidget >> getPreferredSizeMin: minimum_size natural: natural_size [
 		GtkRequisition *natural_size))
 ]
 
+{ #category : #geometry }
+GtkWidget >> getScaleFactor [
```
Comment (estebanlm) [inline src/Mars-Gtk/GtkWidget.class.st]: please, rename this method to just `#scaleFactor`

File: src/Mars-Gtk/GtkWindow.class.st [hunk 2]
```diff
@@ -136,6 +136,13 @@ GtkWindow >> getPositionX: root_x y: root_y [
 	^ self ffiCall: #(void gtk_window_get_position (self, gint *root_x, gint *root_y))
 ]
 
+{ #category : #geometry }
+GtkWindow >> getScaleFactor [
```
Comment (estebanlm) [inline src/Mars-Gtk/GtkWindow.class.st]: same here, just `scaleFactor`. 
The logic is that public api should have pharo standards (yes, there are methods called getX, but you will see they are all on private protocol and are used by other methods with proper name)

**Review by estebanlm — CHANGES_REQUESTED** [review]
just rename those methods :)

File: src/Mars-Gtk/GtkWindow.class.st (hunk 2 above)
Comment (tinchodias) [reply src/Mars-Gtk/GtkWindow.class.st]: done

File: src/Mars-Gtk/GtkWidget.class.st (hunk 1 above)
Comment (tinchodias) [reply src/Mars-Gtk/GtkWidget.class.st]: done!

**Comment by estebanlm** [comment]
oops. I completely forget this, sorry. 
Now, there is another small issue :(

this notation: 

```Smalltalk
^ self ffiCall: #(gint gtk_widget_get_scale_factor #(self))
```

is not quite correct for our conventions. This one (closer to C declaration), is: 

```Smalltalk
^ self ffiCall: #(gint gtk_widget_get_scale_factor (self))
```

If you fix that, you can merge yourself :)

**Comment by tinchodias** [comment]
Done.

These 2 should be fixed too, but not in this PR: `GtkWidget >> dragDestSetFlags:...` and `GtkWidget >> dragDestUnset`.

**Comment by tinchodias** [comment]
I don't have write access to merge it myself :)

**Comment by estebanlm** [comment]
sorry for the delay!

**Pull request merged**